src/vehicle.cpp

```diff
@@ -3955,27 +3955,32 @@ void vehicle::operate_plow(){
 void vehicle::operate_reaper(){
     const tripoint &veh_start = global_pos3();
     for( const int reaper_id : all_parts_with_feature( "REAPER" ) ){
-        const tripoint reaper_pos = veh_start + parts[reaper_id].precalc[0];
-        const int plant_produced =  rng( 1, parts[reaper_id].info().bonus );
-        const int seed_produced = rng(1, 3);
-        const int max_pickup_size = parts[reaper_id].info().size / 20;
-        if( g->m.furn(reaper_pos) == f_plant_harvest ){
-            const item& seed = g->m.i_at(reaper_pos).front();
-            if( seed.typeId() == "fungal_seeds" || seed.typeId() == "marloss_seed" ) {
+        const tripoint reaper_pos = veh_start + parts[ reaper_id ].precalc[ 0 ];
+        const int plant_produced =  rng( 1, parts[ reaper_id ].info().bonus );
+        const int seed_produced = rng( 1, 3 );
+        const int max_pickup_size = parts[ reaper_id ].info().size / 20;
+        if( g->m.furn( reaper_pos ) == f_plant_harvest &&
+            !g->m.i_at( reaper_pos ).empty() ){
```

Comment:
`map::has_items` is a better way of doing that. It doesn't construct the stack object and is more explicit).